libqtile/backend/wayland/core.py

```diff
@@ -391,6 +392,31 @@ def finalize(self) -> None:
         if hasattr(self, "qtile"):
             delattr(self, "qtile")
 
+    def win_from_node(self, node) -> Window | None:
```

Comment:
is it always true that a node must have a window?

Reply:
this code is only called from areas where it is guaranteed we are getting a window from a node. For example when we are iterating through the window tree

Reply:
you make a good point, it can also return internal windows. I changed the signature

Reply:
I guess what I'm really wondering is, if we expect this to always return a window, can we drop the `| None` from the signature and just do the assert in this function, instead of making all the callers do it?

Reply:
I think asserting there is fine. I have added this.